### PR TITLE
Update dissolved date search functionality

### PR DESCRIPTION
**Pull request by ethatcher83**
Changed the dissolved date search fields to be individual input fields
Added validation to check the whole date is completed and valid, show error messages if not

Resolves: BI-10542

### CODE REVIEW

File: src/controllers/advanced-search/search.controller.ts [hunk 1]
```diff
@@ -27,18 +34,18 @@ const route = async (req: Request, res: Response) => {
...
+        return res.render(templatePaths.ADVANCED_SEARCH_RESULTS, { ...errorList, advancedSearchParams, incorporatedFrom, incorporatedTo, selectedStatusCheckboxes, selectedTypeCheckboxes, dissolvedFromDay, dissolvedFromMonth, dissolvedFromYear, dissolvedToDay, dissolvedToMonth, dissolvedToYear, ADV_SEARCH_NUM_OF_RESULTS_TO_DOWNLOAD });
     }
 
     const { companyResource, searchResults } = await getSearchResults(advancedSearchParams, cookies);
     const totalReturnedHits: number = companyResource.hits;
     const totalReturnedHitsFormatted: string = companyResource.hits.toLocaleString();
     const numberOfPages: number = Math.ceil(companyResource.hits / 20);
     const pagingRange = getPagingRange(page, numberOfPages);
-    const partialHref: string = buildPagingUrl(advancedSearchParams, incorporatedFrom, incorporatedTo, dissolvedFrom, dissolvedTo);
+    const partialHref: string = buildPagingUrl(advancedSearchParams, incorporatedFrom, incorporatedTo, dissolvedFromDay, dissolvedFromMonth, dissolvedFromYear, dissolvedToDay, dissolvedToMonth, dissolvedToYear);
```
Comment (VaughanJackson123) [inline src/controllers/advanced-search/search.controller.ts:45]: This `buildPagingUrl` function seems to be taking a lot of parameters!

File: src/controllers/utils/advanced-search-validation.ts [hunk 2]
```diff
@@ -74,43 +85,133 @@ export const advancedSearchValidationRules =
...
-                    if (!isDateFormattedProperly(date)) {
-                        throw Error(DISSOLVED_INVALID_DATE_ERROR_MESSAGE);
+                const dissolvedFromDay = req.query?.dissolvedFromDay as string;
+                const dissolvedFromMonth = req.query?.dissolvedFromMonth as string;
+                const dissolvedFromYear = req.query?.dissolvedFromYear as string;
+
+                // check that one part of the date is not empty and check if day is missing
+                if (isStringNotNullOrEmpty(dissolvedFromYear) || isStringNotNullOrEmpty(dissolvedFromMonth)) {
+                    if (!isStringNotNullOrEmpty(dissolvedFromDay)) {
+                        throw Error(DISSOLVED_FROM_DATE_MUST_INCLUDE_DAY);
```
Comment (VaughanJackson123) [inline src/controllers/utils/advanced-search-validation.ts:97]: Perhaps a named function might be used to replace these duplicate code blocks, make the code briefer and make the purpose and function clearer.

**Review by VaughanJackson123 — APPROVED** [review]

**Review by caismith-ch — APPROVED** [review]

File: src/controllers/advanced-search/search.controller.ts (hunk 1 above)
Comment (ethatcher83) [reply src/controllers/advanced-search/search.controller.ts]: pushed a commit to resolve this

File: src/controllers/utils/advanced-search-validation.ts (hunk 2 above)
Comment (ethatcher83) [reply src/controllers/utils/advanced-search-validation.ts]: separated out into separate functions, thanks

**Pull request merged**